event.go

```diff
@@ -318,6 +318,9 @@ func NewEventFromUntrustedJSON(eventJSON []byte, roomVersion RoomVersion) (resul
 		return
 	}
 
+	if eventJSON, err = sjson.DeleteBytes(eventJSON, "unsigned"); err != nil {
```

Comment:
Won't this nuke `invite_state` etc?

Reply:
It doesn't — the stripped state is sent in the invite federation request and then we add it to the `"unsigned"` section ourselves.